### ISSUE TITLE
Use shim in nuxt report an error when build

### ISSUE THREAD

**Issue by Ttou** [dwightjack/vue-types#149]
nuxt: 2.15.7
vue-types: 4.0.0
repo: [link](https://gitee.com/jh_shot/ssr-legacy-template)

![微信截图_20210726161026](https://user-images.githubusercontent.com/4298437/126955630-e9566fd8-2ff1-4220-b5cc-4bc955905d1f.png)

**Comment by dwightjack** [comment]
Hello, 

I am not completely sure why this is happening. You could try to configure the build following instructions for [webpack 5](https://dwightjack.github.io/vue-types/guide/installation.html#webpack-5).

Also, could you share the part of configuration where you set the shim?

**Comment by Ttou** [comment]
@dwightjack webpack 5 is fine, nuxt is using webpack4

**Comment by Ttou** [comment]
@dwightjack you mean use webpack5 config instead?

**Comment by dwightjack** [comment]
The build behavior is strange. Nuxt still uses webpack 4 but the behaviour reported in the error suggests that the tool is trying to load `vue-types` like webpack 5 does.

I've created a demo installation and I cannot replicate the issue: https://github.com/dwightjack/demo-nuxt-vue-types

Re you using any particular setup (for example: `nuxt-ts`?).

**Comment by Ttou** [comment]
@dwightjack online [example](https://codesandbox.io/s/cool-browser-iisw4)

**Comment by dwightjack** [comment]
@Ttou so the issue might be related to the version of node you're using.

When setting the shim alias try one of the following:

```
'vue-types': 'vue-types/shim'
```

or 

```
'vue-types': require.resolve('vue-types/shim')
```

I will update the docs in the next few days.

**Comment by Ttou** [comment]
another issue, when using custom validator, report error:
![微信截图_20210727173017](https://user-images.githubusercontent.com/4298437/127131219-095f9998-5a2a-4da7-8370-86b7deb66092.png)

**Comment by dwightjack** [comment]
I see.

That's another issue I already found and was going to fix in the next few days.

Sorry for the problem, for now please don't use the shim. I will inform you as soon as the issue is fixed 🙇

**Comment by dwightjack** [comment]
@Ttou I've published a first patch that should fix your issue in `vue-types@4.0.1`. 

Let me know if it works.

**Comment by Ttou** [comment]
@dwightjack fixed
